src/MCPClient/lib/clientScripts/compress_aip.py

```diff
@@ -69,6 +69,17 @@ def compress_aip(job, compression, compression_level, sip_directory, sip_name, s
             'echo program="7z"\; '
             'algorithm="{}"\; '
             'version="`7z | grep Version`"'.format(compression_algorithm))
+    elif program == 'zip':
+        compressed_location = uncompressed_location + ".zip"
+        command = '/usr/bin/7z a -bd -tzip -y -mm={algorithm} -mx={level} -mtc=on -mmt=on "{compressed_location}" "{uncompressed_location}"'.format(
```

Comment:
At a quick glance, it looks like `uncompressed_location` is built from the user-provided name of the SIP, which mean something suitably annoying could escape this quoting.

Maybe `shlex.quote()` could be a safer way to escape those filenames?

(But I acknowledge that breaks the convention established in the rest of the file, so feel free to ignore me!)